File: observer/merged/frontend/src/types/file.ts
// File and Artifact types

import {
  CanonicalStage,
  PipelineId,
  TruthStatus,
  ArtifactStatus
} from './enums';

// Base file from Source B (file_explorer)
export interface FileItem {
  path: string;
  name: string;
  size: number;
  type: string;
  modified: number;  // Unix timestamp
  isDirectory: boolean;
}

// Artifact from Source A (refinery-dashboard)
export interface Artifact {
  id: string;
  name: string;
  projectId: string;
  pipelineId: PipelineId;
  stage: CanonicalStage;
  type: string;
  size: string;
  updatedAt: number;
  tags: string[];
  status: ArtifactStatus;
  isVaulted?: boolean;
  atomClass: string;
  truthStatus: TruthStatus;
}

// UnifiedFile - bridges Artifact and FileItem
export interface UnifiedFile extends Artifact {
  path: string;
  modified: number;
  isDirectory: boolean;
}

// Directory listing response
export interface DirectoryListing {
  path: string;
  files: FileItem[];
  total: number;
}

// File preview response
export interface FilePreview {
  path: string;
  preview: string | null;
  type: string;
  language?: string;
  lineCount?: number;
}

// File metadata
export interface FileMetadata {
  path: string;
  size: number;
  created: number;
  modified: number;
  accessed: number;
  permissions: string;
  owner: string;
  mimeType: string;
}

// Operation request types
export interface PasteRequest {
  files: string[];
  destination: string;
  operation: 'copy' | 'move';
}

export interface DeleteRequest {
  files: string[];
}

export interface CreateFolderRequest {
  path: string;
  name: string;
}

export interface RenameRequest {
  path: string;
  newName: string;
}

// Operation result
export interface OperationResult {
  success: boolean;
  message?: string;
  error?: string;
}

// Search
export interface SearchResult {
  path: string;
  name: string;
  type: string;
  matchContext?: string;
}

export interface SearchResponse {
  query: string;
  results: SearchResult[];
  total: number;
}

File: observer/merged/backend/models/file.py
"""File and Artifact models."""

from pydantic import BaseModel
from typing import List, Optional
from datetime import datetime

from .enums import CanonicalStage, PipelineId, TruthStatus, ArtifactStatus


class FileItem(BaseModel):
    """Base file from file system."""
    path: str
    name: str
    size: int
    type: str
    modified: float  # Unix timestamp
    is_directory: bool = False


class Artifact(BaseModel):
    """Artifact from refinery pipeline."""
    id: str
    name: str
    project_id: str
    pipeline_id: PipelineId
    stage: CanonicalStage
    type: str
    size: str
    updated_at: float
    tags: List[str]
    status: ArtifactStatus
    is_vaulted: Optional[bool] = None
    atom_class: str
    truth_status: TruthStatus


class UnifiedFile(Artifact):
    """Bridges Artifact and FileItem."""
    path: str
    modified: float
    is_directory: bool = False


class DirectoryListing(BaseModel):
    """Directory listing response."""
    path: str
    files: List[FileItem]
    total: int


class FilePreview(BaseModel):
    """File preview response."""
    path: str
    preview: Optional[str] = None
    type: str
    language: Optional[str] = None
    line_count: Optional[int] = None


class FileMetadata(BaseModel):
    """File metadata."""
    path: str
    size: int
    created: float
    modified: float
    accessed: float
    permissions: str
    owner: str
    mime_type: str


class PasteRequest(BaseModel):
    """Copy/move request."""
    files: List[str]
    destination: str
    operation: str = 'copy'  # 'copy' or 'move'


class DeleteRequest(BaseModel):
    """Delete request."""
    files: List[str]


class CreateFolderRequest(BaseModel):
    """Create folder request."""
    path: str
    name: str


class RenameRequest(BaseModel):
    """Rename request."""
    path: str
    new_name: str


class OperationResult(BaseModel):
    """Operation result."""
    success: bool
    message: Optional[str] = None
    error: Optional[str] = None


class SearchResult(BaseModel):
    """Search result item."""
    path: str
    name: str
    type: str
    match_context: Optional[str] = None


class SearchResponse(BaseModel):
    """Search response."""
    query: str
    results: List[SearchResult]
    total: int
